src/replicache-options.ts

```diff
@@ -178,19 +178,22 @@ export interface ReplicacheOptions<MD extends MutatorDefs> {
   pusher?: Pusher;
 
   /**
-   * Allows implementing the underlying storage layer completely in JavaScript.
-   *
-   * @experimental This option is experimental and might be removed or changed
-   * in the future without following semver versioning. Please be cautious.
+   * The license key for Replicache. This parameter is required for Replicache to
+   * function. See https://replicache.dev for how to acquire a license key.
+   *
+   * YOU SHOULD PASS Replicache.TEST_LICENSE_KEY IN AUTOMATED TESTS. It disables license
+   * checks for several minutes. If you pass a normal license key in tests, each test
+   * that instantiates Replicache will attempt to perform a license check against
+   * Replicache's licensing server, potentially increasing your monthly active browser
+   * profile count, slowing the test down, and spamming Replicache's servers.
    */
-  experimentalKVStore?: kv.Store;
+  licenseKey?: string;
```

Comment:
Every property in this interface is optional so I made this one optional as well to just follow the pattern...

Reply:
FWIW, `name` is required.

I think it is OK to keep this optional and default to the `TEST_LICENSE_KEY`

Reply:
I don't want to default to TEST_LICENSE_KEY, I want to default to telling them they need to supply a key. If we default to TEST_LICENSE_KEY then if a customer isn't aware of the license key requirement Replicache starts up just fine and works great for N minutes and then just mysteriously stops working. I think Replicache should work or not, and that you should explicitly configure the license key that you want. Less error prone that way. That is my opinion tho, obviously if folks think differently we can change it.

Reply:
Make this should be required.

Reply:
> mysteriously stops working.

Does it print an error? I was imagining that it would. That would make it less mysterious.

Reply:
I'm not sure how I feel about having to put in the `TEST_LICENSE_KEY` constant. I get Fritz's point, want to try it on for size. If the license key is semantically required, let's mark it required.

However note that we still need to do a runtime check (as name does) because if the caller is javascript (not typescript) nothing prevents them from not passing this param.

Reply:
> Does it print an error? I was imagining that it would. That would make it less mysterious.

To just expand a little on the way I'm looking at it: the customer starts using Replicache and things are going well. Then it stops working. Yes, some percentage of people are going to look in the logs, but that percentage is not 100%, some are going to just going to say 'low quality, this is busted, time to move on'. And of those that DO look in the logs, not 100% are going to see the log line. There's a lot of log spew. 

Now I guess the argument works the other way too: IF we require they take some step like getting a key before using  Replicache themselves then we lose some percentage of people who just dont want to be bothered. 

I dunno how to cut it but it just seems simpler to me mental model wise to implement what we're telling them, which is that using Replicache requires that they get a license.

Reply:
I get where you're coming from. It would be super bad for someone to not receive the info that they need to get a license key, start using Replicache (either with 'test' or empty string if it was defaulted that way), then be confused as to why it randomly stops working.

We want to drive people (via docs, dx, etc) to actually get a license key -- not the reverse. The test license key should only be used for actual automated tests. For development, tire kicking, etc, people should get a license.